wordpress-mobile/WordPress-Android
# Issue/10572 update work manager

Fixes #10572

I'm not able to replicate the issue. This PR updates `androidx.work:` to it's newest stable version (2.4.0).

Release notes are [here](https://developer.android.com/jetpack/androidx/releases/work).

I recommend reviewing this PR by commits:
- https://github.com/wordpress-mobile/WordPress-Android/commit/67ae87054e7f8d94626cfbba696a283c11409389 - v2.1.0 requires java 8 (see [this](https://developer.android.com/jetpack/androidx/releases/work#2.1.0))
- https://github.com/wordpress-mobile/WordPress-Android/commit/6c5d9e579ed70d4a91b03628d1be4fb65ce26665 - `lifecycle-extensions` was deprecated, this PR adds required dependencies explicitly (see [this](https://developer.android.com/jetpack/androidx/releases/lifecycle#declaring_dependencies))
- https://github.com/wordpress-mobile/WordPress-Android/commit/9f3a44583c00e987294053c368f2637787cb16e8 - `ViewModelProviders` were removed with the deprecation of `lifecycle-extensions`
- https://github.com/wordpress-mobile/WordPress-Android/commit/f594c5733e34f65d0fa94e277fbde04f46b76f3d implemented on demand initialization of the Worker, reverted this change in https://github.com/wordpress-mobile/WordPress-Android/commit/eb0c774f2feed6eb364de2c21359933138d10104 since we schedule a periodic worker on apps startup anyway
- https://github.com/wordpress-mobile/WordPress-Android/commit/f1b503925a01b51cf86a1681bdfb9798f782d351 - updated usages of deprecated `WorkManager.getInstance()` with `WorkManager.getInstance(context)`
- 82b0c50cb5cb44a320e1b8e6bcf5a7c71e1efeb7 - use GCMNetworkManager on API <= 22 (see [this](https://developer.android.com/jetpack/androidx/releases/work#2.2.0))


To test - test on a newer API and API <= 22

1. Start the app in airplane mode
2. Create a new post, write a title, press Publish
3. Close the app.
4. Turn off airplane mode.
5. Make sure your device is not on data roaming.
6. Notice the the post gets uploaded almost immediately

------------------------------------
Run ` adb shell dumpsys jobscheduler` (API 23+) and find the periodic task in the list. 
Note: I spent quite some time investigating why `JobInfo` for the periodic task doesn't contain 
```
PERIODIC: interval=+8h0m0s0ms flex=+6h0m0s0ms
```
It does contain this line when running the app before this PR. I haven't find any information in docs about this change, but I tried [switching the interval for the task to "15 minutes" and flex to "5 minutes"](https://github.com/wordpress-mobile/WordPress-Android/blob/94a352cb0cd173cfb3ff9cc26650cfe910783a40/WordPress/src/main/java/org/wordpress/android/util/UploadWorker.kt#L80) and verified it runs periodically.

PR submission checklist:
